foreign_cc/built_tools/cmake_build.bzl

```diff
@@ -1,14 +1,40 @@
-""" Rule for building CMake from sources. """
+""" Macro for building CMake from sources. """
 
-load("//foreign_cc:defs.bzl", "configure_make")
+load("//foreign_cc:defs.bzl", "cmake_variant", "configure_make")
 
 def cmake_tool(name, srcs, **kwargs):
+    """Macro for building CMake
+
+    Args:
+        name: A unique name for this target
+        srcs: The target containing the build tool's sources
+        **kwargs: Remaining args
+    """
     tags = ["manual"] + kwargs.pop("tags", [])
 
-    configure_make(
+    native.alias(
         name = "{}.build".format(name),
+        actual = select({
+            ":msvc_compiler": "{}_msvc".format(name),
+            "//conditions:default": "{}_default".format(name),
+        }),
+    )
+
+    cmake_variant(
+        name = "{}_msvc".format(name),
+        # to prevent errors with incompatible _WIN32_WINNT in cmlibarchive
+        # override NTDDI_VERSION to match _WIN32_WINNT set in the default cc_toolchain
+        copts = ["/D NTDDI_VERSION=0x06010000"],
```

Comment:
I'm not familiar enough with Windows to understand why this is needed but going to trust you on this one!

Reply:
Yeah I would very much have liked not to include this but there is definitely a error with this package and the default toolchain.